src/cli/send/junitUtils.spec.ts

```diff
@@ -0,0 +1,216 @@
+import { transformToJunit } from './junitUtils';
+
+describe('transformToJunit', () => {
+  it('should output valid request', () => {
+    const result = transformToJunit({
+      _meta: {
+        version: 'test',
+      },
+      requests: [
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test',
+          duration: 1001,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+      ],
+      summary: {
+        totalRequests: 1,
+        successRequests: 1,
+        disabledRequests: 0,
+        failedRequests: 0,
+        totalTests: 1,
+        successTests: 1,
+        failedTests: 0,
+      },
+    });
+    expect(result).toBe(
+      `
+<testsuites errors="0" disabled="0" failues="0">
+  <testsuite name="test.http" tests="1" failures="0" skipped="0" package="." time="1.001" file="test.http">
+    <properties>
+      <property name="name" value="test"/>
+    </properties>
+    <testcase name="test" classname="test.http" assertions="1" time="1.001"/>
+  </testsuite>
+</testsuites>
+    `.trim()
+    );
+  });
+
+  it('should output multiple requests', () => {
+    const result = transformToJunit({
+      _meta: {
+        version: 'test',
+      },
+      requests: [
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test',
+          duration: 1001,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+        {
+          fileName: 'test.http',
+          disabled: false,
+          name: 'test2',
+          duration: 1002,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+        {
+          fileName: 'test2.http',
+          disabled: false,
+          name: 'other',
+          duration: 1002,
+          summary: {
+            totalTests: 1,
+            successTests: 1,
+            failedTests: 0,
+          },
+        },
+      ],
+      summary: {
+        totalRequests: 1,
+        successRequests: 1,
+        disabledRequests: 0,
+        failedRequests: 0,
+        totalTests: 1,
+        successTests: 1,
+        failedTests: 0,
+      },
+    });
+    expect(result).toBe(
+      `
+<testsuites errors="0" disabled="0" failues="0">
+  <testsuite name="test.http" tests="2" failures="0" skipped="0" package="." time="2.003" file="test.http">
+    <properties>
```

Comment:
the properties node here is a bit strange. I would expect that a testsuite has at most 1 properties node but the properties node can have multiple property child nodes. It seems like we are duplicating the node without nesting into a test case to try and imply association based on position of properties node in relation to the test case in the output,

Reply:
testcase unfortunately has no properties element. I would now use the same hack as [playwright](https://github.com/microsoft/playwright/blob/c3b533d8341d72d45b7296c7a895ff9fe7d8ff3b/packages/playwright/src/reporters/junit.ts#L150-L170) and just put the properties below testcase.